programs/uxd/src/instructions/compute_depositories_targets.rs

```diff
@@ -0,0 +1,261 @@
+use crate::error::UxdError;
+use crate::state::CredixLpDepository;
+use crate::state::MercurialVaultDepository;
+use crate::utils::checked_convert_u128_to_u64;
+use crate::utils::compute_amount_fraction_floor;
+use crate::validate_is_program_frozen;
+use crate::Controller;
+use crate::BPS_UNIT_CONVERSION;
+use crate::CONTROLLER_NAMESPACE;
+use crate::CREDIX_LP_DEPOSITORY_NAMESPACE;
+use crate::MERCURIAL_VAULT_DEPOSITORY_NAMESPACE;
+use anchor_lang::prelude::*;
+
+#[derive(Accounts)]
+pub struct ComputeDepositoriesTargets<'info> {
+    /// #1 Permissionless IX that can be called by anyone
+    #[account(mut)]
+    pub payer: Signer<'info>,
+
+    /// #2 The top level UXDProgram on chain account managing the redeemable mint
+    #[account(
+        mut,
+        seeds = [CONTROLLER_NAMESPACE],
+        bump = controller.load()?.bump,
+        constraint = controller.load()?.registered_mercurial_vault_depositories.contains(&mercurial_vault_depository.key()) @UxdError::InvalidDepository,
+        constraint = controller.load()?.registered_credix_lp_depositories.contains(&credix_lp_depository.key()) @UxdError::InvalidDepository,
+    )]
+    pub controller: AccountLoader<'info, Controller>,
+
+    /// #3 The first known active mercurial vault depository
+    #[account(
```

Comment:
In the case we create a new depository, this permissionless ix could be called and mess around depositories targets.

We could:
- Use dynamic `remaining_accounts` to pass all registered mercurial vaults and credix vaults depositories.
- Enforce known mercurial/credix depositories `PublicKey`.
- Disable new mercurial/credix depository creation.
- Keep it as it is and know internally that creating a new depository may break rebalancing in a bad way (and don't do that. If we need to add a new repo, need to update the tx first).

Reply:
yeah definitly something we will have to keep in mind if we add new depositories.

Im not sure i understand what you mean by "breaking" the rebalancing.

If we create a new depository, its target will just be 0 until the compute_target IX knows about it, so it will just be automatically allocated to 0%, which kinda make sense if we just add depositories without specifying the rules around it in the code